My sequences Page

denier-epnak opened the pull request:
Define 'My courses' page with some necessary extensions:

- local plugin padplusextensions to host any customisation which can not be made in theme (global navigation, new renderer)
- Icon system extension in theme

denier-epnak committed:
refactor: define local plugin padplusextensions and extract navigation node 'all categories'

Local plugin padplusextensions should host any PAD+ customisation which can not be easily overridden in PAD+ theme.

For example we now declare the 'all categories' navigation node in function local_padplusextensions_extend_navigation, instead of adding it as a side-effect in columns2 sidebar.

denier-epnak committed:
refactor: override Font Awesome icon system in PAD+ theme

This is a cleaner way to extend/override the Moodle icon system which is used in sidemenu for example.

denier-epnak committed:
Define 'my courses' page at my/courses.php URL

- create a new entry point in Moodle page rooting
- create a custom renderer in PAD+ local extensions
- create a custom log eventmycoursesviewed
- extend global navigation with node 'mycoursespage' to be used in sidemenu / breadcrumb (only for authenticated users)

mantecajc commented on local/padplusextensions/lang/en/local_padplusextensions.php:
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comme tu l'as placé dans le dossier local, ce plugin est-il uniquement disponible en local ?

J'ai du mal à voir à quoi sert ce plugin.

denier-epnak replied:
Non le type 'local plugin' est un peu un type de plugin par défaut https://docs.moodle.org/dev/Local_plugins

Le truc c'est que ça permet de surcharger la navigation, ce qui a priori n'est pas possible dans un thème (on avait essayé je crois)
https://docs.moodle.org/dev/Navigation_API#Local_Plugins
https://github.com/moodle/moodle/blob/MOODLE_36_STABLE/lib/navigationlib.php#L1458

De même pour déclarer un nouveau renderer (pareil, a priori pas trop possible dans un thème, qui est conçu pour "surcharger" l'existant)

Bref, c'est pour ce qui ne fit pas dans le thème.

denier-epnak replied:
A propos de la navigation, c'est aussi le seul moyen propre que j'ai trouvé pour étendre celle-ci et faire en sorte que la page "Mes séquences" apparaisse correctement dans le fil d'Ariane. La construction de ce fil d'Ariane reste un mystère pour moi (surtout les règles qui décident de comment construire ce fil en fonction du contexte)

mantecajc replied:
Ah ok je vois !

mantecajc commented on my/courses.php:
```diff
@@ -0,0 +1,48 @@
+<?php
```
c'est possible de créer cette page directement dans le thème padplus ?

denier-epnak replied:
dans ce cas on aurait une URL comme theme/padplus/mycourses.php, ce qui est un peu bizarre sémantiquement. J'ai suivi la convention Moodle 4, où la page "mes séquences" sera à cette adresse my/courses.php

mantecajc replied:
Ah d'accord je vois ! C'est vrai que si nous n'écrasons pas le code moodle mais ajoutons de nouveaux fichiers, il ne devrait pas y avoir de problème de le faire hors le theme

denier-epnak commented on course/externallib.php:
```diff
@@ -3869,14 +3869,15 @@ function($favourite) {
             if (in_array($course->id, $favouritecourseids)) {
                 $isfavourite = true;
             }
-            $exporter = new course_summary_exporter($course, ['context' => $context, 'isfavourite' => $isfavourite]);
```
Missing PADPLUS mark

denier-epnak committed:
Display regular pagination for myoverview block

This commit contains a fix/hack in PageContentFactory buildPagingBarTemplateContextKnownLength method, to account for itemsPerPage config object in myoverview context. Specifically, it accepts itemsPerPage as {value: <number>, active: <boolean>} parameter so that it can compute the number of pages.

denier-epnak committed:
Display 'shared resource' or 'catalog' tags in course card and total
course count

denier-epnak committed:
Customize 'add course/workshop' button on leaf category page and refactor padplus lib

- extract category_belongs_to_workshop/category_belongs_to_catalog predicates for reuse

denier-epnak committed:
Add Catalog filter in myoverview block

Please note this only works by surgically hacking core modules:
- course/lib|externallib: provide a new value COURSE_CATALOG_PADPLUS='catalog' for get_enrolled_courses_by_timeline_classification webservice calls as well as custom function course_filter_courses_in_catalog to fulfill the feature
- blocks/myoverview/lib|settings: enable BLOCK_MYOVERVIEW_GROUPING_CATALOG_PADPLUS='catalog' to be used as user preference for course grouping
- blocks/myoverview/main: conditionnally render catalog filter based on user catalog access

denier-epnak committed:
Various fix/refactor for 'my courses' page and dashboard

- fix myoverview style on dashboard and my courses page
- extract and use iconbutton mixin for favourite button in course
card/item
- extract and use pagination mixin on category page/myoverview block
- split styles between block_myoverview component, coursecards
component, and dashboard pages
